### DISCUSSION TITLE
12.0.1

### DISCUSSION THREAD

**Discussion by SubJunk** [UniversalMediaServer/UniversalMediaServer#3687] [General]
We are so excited about this complete rewrite of our GUI!
It's built using modern tooling and will let us develop far more quickly than the old one did. The old one is still there for those who want it, though.

When you start UMS, it will open in your web browser. The first time it runs, it will offer you the chance to create a user, or to disable authentication:

![image](https://user-images.githubusercontent.com/584228/207146703-b7b0045c-ec69-45ed-bab3-d344bfb4576e.png)

We think it's a good idea to create an admin account unless you are sure your network is private, but the choice is yours.

You can change it at any time, and you can create different users who have different permissions, and different user groups too. We will expand that functionality over time to include more permissions. Let us know your ideas for this, since that will affect which features we prioritize.

Another change is that the Shared Content area has been unified, where before it was split between local and web content, now they are in the same list.

![image](https://user-images.githubusercontent.com/584228/207146733-c1a4fb43-5b90-4297-9cb9-40c461f4c0b9.png)

We are hoping to get some feedback about this. Please let us know if you find any bugs or think of improvements you'd like to see.

I would like to thank the massive efforts of @SurfaceS and @js-kyle on this version. It's rare with software to do a full rewrite. Usually limitations are accepted and we do our best with what we have, because it's a lot of hard - and at times tedious - work to start from scratch, but we have finally done it after years of wanting to.

There are so many more changes in this release, including hundreds of bugfixes and performance improvements. See the full changelog below. Note that we did not release 12.0.0, because we found a serious bug after it was tagged, so 12.0.1 is the first actual release.

### General:
- Added a new web settings interface, which is now the supported way to add content and change settings
- Added a new web player interface, for a faster, more responsive and accessible web player
- Added shutdown computer option to the Server Settings folder
- Added user auth to web interface
- Added user groups to web interface
- Unified the Shared Content area in both Java and web interfaces
- Local database speed improvements
- Improved UPnP/DLNA support
- Consolidated configuration files, with automatic migration to the new format
- Fixed TV series never being found locally by IMDb ID
- Fixed duplicate API requests
- Fixed profile support on Docker
- Fixed broken translations on startup language selection
- Fixed not detecting Playlist additions/changes during scans
- Fixed database update/drop
- Hundreds of bugfixes and performance tweaks
### Renderers:
- Support Sony Network Speakers (thanks, scanf!)
### Translation updates via Crowdin
- Afrikaans (14%) (thanks, Eugene Trumpelmann!)
- Bulgarian (83%) (thanks, ruraru!)
- Catalan (88%) (thanks, Toni Grau i Quellos!)
- Czech (100%)
- Danish (99%) (thanks, GurliGebis and NCAA!)
- German (93%) (thanks, pipin!)
- Finnish (99%) (thanks, Esko Gardner!)
- Italian (96%) (thanks, Oscar Zambotti!)
- Korean (100%) (thanks, VenusGirl!)
- Polish (100%) (thanks, Karol Szastok!)
- Portuguese (99%) (thanks, mariopinto!)
- Portuguese (Brazilian) (99%) (thanks, Mauro.A!)
- Russian (82%) (thanks, Максим Мухачев!)
- Slovak (81%) (thanks, Dušan!)
- Spanish (92%) (thanks, Pablo Camacho!)
- Turkish (100%) (thanks, Burak Yavuz!)
### Dependencies
- Bump Java to 17.0.5
- Bump logback-version from 1.4.3 to 1.4.4
- Update Apache Commons Collections to 4.4
- Update com.sun.xml.messaging.saaj to 3.0.0
- Update dependency com.fasterxml.jackson.core:jackson-databind to v2.14.1
- Update dependency com.github.junrar:junrar to v7.5.4
- Update dependency com.github.oshi:oshi-core to v6.3.2
- Update dependency com.github.spotbugs:spotbugs-maven-plugin to v4.7.3.0
- Update dependency com.google.code.gson:gson to v2.10
- Update dependency com.ibm.icu:icu4j to v72
- Update dependency com.puppycrawl.tools:checkstyle to v10.4
- Update dependency net.coobird:thumbnailator to v0.4.18
- Update dependency org.slf4j:slf4j-api to v2.0.5
- Update dependency pako to v2.1.0
- Update dependency video.js to v7.21.1
- Update dependency videojs-contrib-quality-levels to v2.2.1
- Update logback-version to v1.4.5
- Update MediaInfo to 22.09
- Update twelvemonkeys-imageio-version to v3.9.4
- Removed assertj-core

<hr /><em>This discussion was created from the release <a href='https://github.com/UniversalMediaServer/UniversalMediaServer/releases/tag/12.0.1'>12.0.1</a>.</em>

**Comment by ExSport** [comment]
Hi guys
Great progress! A long time ago I tested UMS last time so I was cusious what is new:)
I spotted some differences from past (Win11) + adding additional thoughts.
- there was no question about AviSynth during the installation
- I started the Wizard and it offers choosing own shared folder and if not it will do it automatically per OS. It is not true as I didn't skip it, chose one but still the default ones were added anyway :) (Wording seemed to me it will happen only when no custom folder is chosen)
- If I tried to delete them later and saved it (WebGui) but the removed ones remained until a did a manual page refresh
- I tried to show log via web but it was empty. I had to choose page (1) to make it visible
- Nice uniform media sharing config. I tried to rename shared folder by EDIT button but as TYPE can't be edited I had to delete the shared folder and created new one VIRTUAL_FOLDER with same path but custom name. It will be great to rename shared folders directly via EDIT or offer changing current TYPE or make them all virtual? It will be more user friendly.
- Clicking on magnifier icon to rescan folders, there is no feedback it was accepted, it is still running or it finished already.
- When choosing folder to share, it offers "Select" to choose it but if I went inside it (lowest depth), there is no option to choose it and you have to go back. It would be nice to offer "Select" also in the lowest deep.
- I tried to browse some pictures. If there are more of them, I miss the option to browse to the next/previous one directly from the opened photo without the need to go back and choose another one or using keyboard "arrows" to go to the NEXT/PREVIOUS.
- I tried to see if picture is original jpeg or transcoded one and I see the file extension is .JFIF. I expect it is the original file, unmodified one, only renamed?
- As opened picture includes EXIF info, it will be very nice to make such info available (show/hide) in the browser.
- When picture resolution is bigger than display, it will be great the picture will be shrinked to the visible area and expanded to the original resolution on click? If I right-click on the picture and open it in the new window, Edge/Chrome will do this (shrinked+zoom option). It seems UMS shrinks it horizontally only so user must scroll down to see whole picture. I understand it will differ per user preference and maybe hardly doable at all but for "some people" very nice to have it :) (by default shrinked horizontally and vertically, on click shrinked horizontally only, on other click to enlarge to the original resolution but this one can be done by opening it in separate window, manually, when needed).
- Browsing video clips there seems to be similar issue as with pictures. When video clip is opened, I see very blurry picture (expanded small thumb) without any controls except PLAY. Until I start the video, there is no NEXT/PREVIEW icons visible so it is impossible to browse them without PLAYING file or going back to the folder itself.
- no more time to test it further... :)

Anyway guys, great work! Very responsive web browsing! Tested locally but anyway, very surprised with browser player!
Thank you guys 👍

**Reply by SubJunk** [reply]
@ExSport great feedback as always, thank you! We are working hard to resolve these bugs with saving content now
